src/gui/folder.cpp

```diff
@@ -533,9 +584,26 @@ void Folder::downloadVirtualFile(const QString &_relativepath)
 
 void Folder::setUseVirtualFiles(bool enabled)
 {
-    _definition.useVirtualFiles = enabled;
-    if (enabled)
+    if (enabled && _definition.virtualFilesMode == Vfs::Off) {
+        _definition.virtualFilesMode = bestAvailableVfsMode();
+
+        _vfs = createVfsFromPlugin(_definition.virtualFilesMode, this);
+        startVfs();
+
         _saveInFoldersWithPlaceholders = true;
+    }
+    if (!enabled && _definition.virtualFilesMode != Vfs::Off) {
+        ENFORCE(_vfs);
+
+        // TODO: Must wait for current sync to finish!
+        SyncEngine::wipeVirtualFiles(path(), _journal, _vfs);
+
+        _vfs->stop();
+        _vfs->unregisterFolder();
+        delete _vfs;
```

Comment:
and _vfs is left dangling.  Is that not a problem?
Maybe use QScopedPointer for vfs